### PR TITLE
Update to OData Library v7.6, fix the failing test cases

### PR DESCRIPTION
**Pull request by xuzhg**
Update to OData Library v7.6, fix the failing test cases

1. Navigation property binding sorted by the path string
2. Context Uri changes for $expand by add parathesis
3. Complex type parameter supports using ODataResourceValue
4. etc.

### Description

*Briefly describe the changes of this pull request.*

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: sln/WebApiOData.Performance.Official.sln [hunk 1]
```diff
@@ -19,12 +19,12 @@ Global
 		Release|Any CPU = Release|Any CPU
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{F7B5A5D5-0B2F-493B-94B2-039B881FD4D3}.CodeAnalysis|Any CPU.ActiveCfg = Release|Any CPU
-		{F7B5A5D5-0B2F-493B-94B2-039B881FD4D3}.CodeAnalysis|Any CPU.Build.0 = Release|Any CPU
-		{F7B5A5D5-0B2F-493B-94B2-039B881FD4D3}.Debug|Any CPU.ActiveCfg = Release|Any CPU
-		{F7B5A5D5-0B2F-493B-94B2-039B881FD4D3}.Debug|Any CPU.Build.0 = Release|Any CPU
-		{F7B5A5D5-0B2F-493B-94B2-039B881FD4D3}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{F7B5A5D5-0B2F-493B-94B2-039B881FD4D3}.Release|Any CPU.Build.0 = Release|Any CPU
+		{F7B5A5D5-0B2F-493B-94B2-039B881FD4D3}.CodeAnalysis|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (KanishManuja-MS) [inline sln/WebApiOData.Performance.Official.sln]: Are these changes required to sln files? Shouldn't the mode remain as Release?

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/App.config [hunk 2]
```diff
@@ -54,6 +54,18 @@
         <bindingRedirect oldVersion="0.0.0.0-2.7.41101.299" newVersion="2.7.41101.299" />
       </dependentAssembly>
 
+      <dependentAssembly>
+        <assemblyIdentity name="Microsoft.Data.Edm" publicKeyToken="31bf3856ad364e35" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-5.8.4.0" newVersion="5.8.4.0" />
```
Comment (KanishManuja-MS) [inline test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/App.config]: >bindingRedirect oldVersion="0.0.0.0-5.8.4.0" newVersion="5.8.4.0 [](start = 9, length = 64)

Why is this change needed?

Comment (xuzhg) [reply test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/App.config]: That‘s from VS automatically.

File: sln/WebApiOData.Performance.Official.sln (hunk 1 above)
Comment (xuzhg) [reply sln/WebApiOData.Performance.Official.sln]: That's from VS automatically, I revert it.

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/DollarLevels/DollarLevelsTest.cs [hunk 3]
```diff
@@ -56,15 +56,13 @@ public async Task LevelsWithInvalidValue(string query, string errorMessage)
...
-        [InlineData("$expand=Manager($levels=0)", 
-            "")]
+        [InlineData("$expand=Manager($levels=3)", "$expand=Manager($expand=Manager($expand=Manager))")]
+        [InlineData("$expand=Manager($levels=0)", "")]
         [InlineData("$expand=Manager($levels=2;$expand=DirectReports($levels=2))",
             "$expand=Manager($expand=Manager($expand=DirectReports($expand=DirectReports)),DirectReports($expand=DirectReports))")]
         [InlineData("$expand=Manager($select=ID;$expand=DirectReports($levels=2;$select=Name))",
-            "$expand=Manager($select=ID;$expand=DirectReports($expand=DirectReports($select=Name);$select=Name))")]
-        public async Task LevelsWithValidNumber(string originalQuery, string expandedQuery) 
+            "$expand=Manager($select=ID;$expand=DirectReports($expand=DirectReports($select=Name);$select=Name,DirectReports))")]
```
Comment (KanishManuja-MS) [inline test/E2ETest/Microsoft.Test.E2E.AspNet.OData/DollarLevels/DollarLevelsTest.cs]: >DirectReports [](start = 111, length = 13)

Why is this additional DirectReports appended? The original query does not select the DirectReports.

File: test/PerfTest/WebApiPerformance.Service/WebApiPerformance.Service.csproj [hunk 4]
```diff
@@ -22,22 +22,11 @@
     <IISExpressUseClassicPipelineMode />
     <UseGlobalApplicationHostFile />
     <TestType Condition=" '$(TestType)' == '' ">Private</TestType>
+    <Use64BitIISExpress />
   </PropertyGroup>
```
Comment (KanishManuja-MS) [inline test/PerfTest/WebApiPerformance.Service/WebApiPerformance.Service.csproj]: Can remove?

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/DollarLevels/DollarLevelsTest.cs (hunk 3 above)
Comment (xuzhg) [reply test/E2ETest/Microsoft.Test.E2E.AspNet.OData/DollarLevels/DollarLevelsTest.cs]: It looks required so far from the logical of ProcessLevel(..).

File: test/PerfTest/WebApiPerformance.Service/WebApiPerformance.Service.csproj (hunk 4 above)
Comment (xuzhg) [reply test/PerfTest/WebApiPerformance.Service/WebApiPerformance.Service.csproj]: That‘s from VS automatically. I revert it
